### ISSUE TITLE
polkadot.js UI transaction pending bubble not leaving after transaction is done

### ISSUE THREAD

**Issue by noot** [ChainSafe/gossamer#1569]
## Describe the bug

- in the polkadot.js UI, if you send a transaction of any type, the blue pending bubble in the top right corner that shows the transaction you sent doesn't go away when the transaction has been included
- this bubble should leave after the transaction is included
- this probably indicates that the UI is not properly getting notified when a transaction is included

## Possible Solution

- maybe related to `submitAndWatchExtrinsic`?

## To Reproduce
Steps to reproduce the behaviour:

1. follow instructions here https://chainsafe.github.io/gossamer/integrate/connect-to-polkadot-js/
2.  send any sort of transaction

**Comment by EclesioMeloJunior** [comment]
When we send a new transaction we save the request extrinsic bytes into the property `extrinsic` from the listener `ExtrinsicSubmitListener` ([here](https://github.com/ChainSafe/gossamer/blob/eef03c6bdcf4cb358ef597924ef82bcb45b0427f/dot/rpc/subscription/websocket.go#L276))

I have noticed that when the listener receives an imported block the value of `block.Body.AsExtrinsics()` doesn't match with the one registered on `ExtrinsicSubmitListener.extrinsic` ([here](https://github.com/ChainSafe/gossamer/blob/eef03c6bdcf4cb358ef597924ef82bcb45b0427f/dot/rpc/subscription/listeners.go#L170))

I suspected that it would be other blocks that were arriving, so I started to log the extrinsic and the hash of the block, and in fact, the transaction-related block doesn't match the extrinsic registered in the listener

1. I create a new transaction then I logged this transaction extrinsic
![init-submitExtrinsicWatch](https://user-images.githubusercontent.com/17255488/118317538-3293bf80-b4c6-11eb-8a90-04e710f0f60e.png)


2. When the listener catch the imported block I logged some infos like, extrinsic of the block and the block hash, as we can see the function `block.Body.AsExtrinsics()` returns a set of extrinsics to compare with the one stored in the listener, but no one can be matched.
![imported-block-arrives](https://user-images.githubusercontent.com/17255488/118317550-37587380-b4c6-11eb-97b0-c922b8fce0d1.png)

3. The block number 4 is the block with the transaction data
![block-number 4](https://user-images.githubusercontent.com/17255488/118317629-4ccd9d80-b4c6-11eb-9ff2-c5fd71ba1c29.png)
![block data](https://user-images.githubusercontent.com/17255488/118317635-4e976100-b4c6-11eb-9481-2e12fdba9cbb.png)

**Comment by EclesioMeloJunior** [comment]
It was difficult lol, but I think I could find out where the extrinsic is being modified

The extrinsics appear OK in the function `buildBlockExtrinsics`

![image](https://user-images.githubusercontent.com/17255488/118333738-3d584f80-b4da-11eb-9f04-a66342e68e0f.png)

But it is modified when the function `buildBlock` calls `extrinsicsToBody` ([here](https://github.com/ChainSafe/gossamer/blob/86acc4342226248eba7ebce00a5cbceeda62cce9/lib/babe/build.go#L100)). The `extrinsicsToBody` function seems decode the original extrinsic and then append the block inherents to the decoded extrinsic.

![image](https://user-images.githubusercontent.com/17255488/118333789-54973d00-b4da-11eb-853f-caad87a3b653.png)

So basically, I think that we need another property to check if the arrived block is what the listener expects, or add a new property called `originalExtrinsic` and use this field to compare. @noot @dutterbutter

**Comment by noot** [comment]
@EclesioMeloJunior it looks like the "checking extrinsic received" just need to be scale decoded, can you decode the extrinsic before checking for equality in listener.go?

**Comment by EclesioMeloJunior** [comment]
closed with PR #1592
